py-morse-code/morse_code.py

```diff
@@ -0,0 +1,69 @@
+from node import Node
+
+
+class MorseCode:
+
+    _code_map = {
+        " ": "  ",
+        "a": ".- ",
+        "b": "-... ",
+        "c": "-.-. ",
+        "d": "-.. ",
+        "e": ". ",
+        "f": "..-. ",
+        "g": "--. ",
+        "h": ".... ",
+        "i": ".. ",
+        "j": ".--- ",
+        "k": "-.- ",
+        "l": ".-.. ",
+        "m": "-- ",
+        "n": "-. ",
+        "o": "--- ",
+        "p": ".--. ",
+        "q": "--.- ",
+        "r": ".-. ",
+        "s": "... ",
+        "t": "- ",
+        "u": "..- ",
+        "v": "...- ",
+        "w": ".-- ",
+        "x": "-..- ",
+        "y": "-.-- ",
+        "z": "--.."
+    }
+
+    @staticmethod
+    def convert_to(value):
+        result = ""
+
+        if value != "":
+            for c in value:
+                result += MorseCode._code_map[c.lower()]
+
+        return result.rstrip()
+
+    @staticmethod
+    def convert_from(value):
+        result = ""
+
+        value = value + " "
+
+        words = value.split("  ")
+
+        for w in words:
```

Comment:
Since the loop will create a new string every iteration, it'd probably be more optimal/portable to build a list then concat it together.


```python
results = []
for w in words:
  results.append(MorseCode._convert_from_word(w))

return ' '.join(results)
```

a more pythonic way would be

```python
results = [
  MorseCode._convert_from_word(w)
  for w in words
]
```